File: Practice/src/coding_basic_training/PerformDifferentOperationsBasedOnLengthOfArray.java
package coding_basic_training;

public class PerformDifferentOperationsBasedOnLengthOfArray {
//	배열의 길이에 따라 다른 연산하기
	public int[] solution(int[] arr, int n) {
		for (int i = 0; i < (arr.length + 1) / 2; i++) {
			arr[arr.length % 2 == 0 ? i * 2 + 1 : i * 2] += n;
		}
		return arr;
	}
}

File: Practice/src/coding_basic_training/PerformDifferentOperationsBasedOnLengthOfArray.js
function solution(arr, n) {
    return arr.map(
        (v, i) =>
            v +
            (arr.length % 2 == 0 && i % 2 == 1
                ? n
                : arr.length % 2 == 1 && i % 2 == 0
                ? n
                : 0)
    );
    // return arr.map((v, i) => (arr.length % 2 ^ i % 2 ? v + n : v));
}
